awx/main/tasks/receptor.py

```diff
@@ -0,0 +1,318 @@
+# receptor.py
+# Python
+from base64 import b64encode
+from collections import namedtuple
+import concurrent.futures
+import socket
+import sys
+import threading
+import time
+import yaml
+
+# Django
+from django.conf import settings
+
+
+# Runner
+import ansible_runner
+import ansible_runner.cleanup
+from awx.main.utils.common import (
+    deepmerge,
+    parse_yaml_or_json,
+    cleanup_new_process,
+)
+
+
+# AWX
+from awx.main.utils.receptor import get_receptor_ctl, get_conn_type, get_tls_client
+from awx.main.tasks import logger
```

Comment:
In these cases, I would advocate defining the logger in _this_ file, and using the "awx.main.tasks.receptor" logger. I welcome any other opinions.